File: CppPractice/Project1/Project1/Leetcode78.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;


class Solution {
public:
    vector<vector<int>> ans;
    void helper(vector<int>& nums, vector<int>& temp,int start)
    {

        if (find(ans.begin(), ans.end(), temp) == ans.end())
        {
            ans.push_back(temp);
          
        }
        for (int i = start; i < nums.size(); i++)
        {
            temp.push_back(nums[i]);
            helper(nums, temp, i + 1);
            temp.pop_back();
        }
        return;

    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> temp;
        int start = 0;
        helper(nums, temp,start);
        return ans;

    }
};

File: CSharpPractice/LeetcodePractice/Leetcode78.cs
public class Solution {
    public int[] pnums;
    public List<IList<int>> ans =  new List<IList<int>>();
    public int n=0;
    public void helper(int start,List<int> temp)
    {
        ans.Add(new List<int>(temp));
        for(int i=start;i<n;i++)
        {
            temp.Add(pnums[i]);
            helper(i+1,temp);
            temp.RemoveAt(temp.Count-1);
        }
    }

    public IList<IList<int>> Subsets(int[] nums) 
    {
        n = nums.Length;
        pnums = nums;
        List<int> temp=new List<int>();
        helper(0,temp);
        return ans;
    }
}